src/main/java/io/zksync/sdk/musig/SchnorrMusigSigner.java

```diff
@@ -0,0 +1,183 @@
+package io.zksync.sdk.musig;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import io.zksync.sdk.musig.entity.AggregatedCommitment;
+import io.zksync.sdk.musig.entity.AggregatedPublicKey;
+import io.zksync.sdk.musig.entity.AggregatedSignature;
+import io.zksync.sdk.musig.entity.Commitment;
+import io.zksync.sdk.musig.entity.MusigSigner;
+import io.zksync.sdk.musig.entity.Precommitment;
+import io.zksync.sdk.musig.entity.Signature;
+import io.zksync.sdk.musig.exception.SchnorrMusigException;
+import io.zksync.sdk.musig.utils.Bytes;
+
+public class SchnorrMusigSigner {
+
+    private final SchnorrMusigNative musig;
+    private final MusigSigner signer;
```

Comment:
We should probably store public keys used to initialize the signer as a field, instead of passing them to methods as arguments.

Reply:
Basically `verify` function doesn't need in instantiated signer. I think here in signer class I will remove parameter to use public keys from signer instance and will make duplicate of the `verify` with public key(s) parameter in SchnorrMusig class as common. Guess it's the best option